projects/core-api/src/main/java/dan200/computercraft/api/filesystem/IMount.java

```diff
@@ -41,6 +41,17 @@ public interface IMount {
      */
     boolean isDirectory(String path) throws IOException;
 
+    /**
+     * Returns whether a file with a given path is read-only or not.
+     *
+     * @param path A file path in normalised format, relative to the mount location. ie: "programs/myprograms".
+     * @return If the file exists and is read-only.
+     * @throws IOException If an error occurs when checking whether the file is read-only.
+     */
+    default boolean isReadOnly(String path) throws IOException {
+        return true;
+    }
+
```

Comment:
I think it probably makes more sense to move this to `IWritableMount` and then default to returning `false`.